lib/home_page/view_page.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+import '../constants/strings.dart' as strings;
+import '../movie.dart';
+import '../movie_category.dart';
+import '../movie_repository.dart';
+import 'home_categories_list.dart';
+import 'upcoming_movies_list.dart';
+
+class ViewPage extends StatefulWidget {
+  const ViewPage({super.key});
+
+  @override
+  State<ViewPage> createState() => _ViewPageState();
+}
+
+class _ViewPageState extends State<ViewPage>
+    with AutomaticKeepAliveClientMixin {
+  final Future<List<Movie>> movies = MovieRepository().getMovies();
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return FutureBuilder(
+      future: movies,
+      builder: (BuildContext context, AsyncSnapshot<List<Movie>> snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const CircularProgressIndicator();
+        }
+        if (snapshot.hasData) {
+          List<Movie> movies = snapshot.data as List<Movie>;
```

Comment:
I think the typeCast is not necessary, you can change it to: `var movies = snapshot.data`